# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/04/22 13:34:58-07:00 akpm@osdl.org 
#   [PATCH] s390: 3270 device driver.
#   
#   From: Martin Schwidefsky <schwidefsky@de.ibm.com>
#   
#   3270 device driver changes:
#    - Add NULL pointer checks.
# 
# drivers/s390/char/raw3270.c
#   2004/04/22 01:40:32-07:00 akpm@osdl.org +3 -1
#   s390: 3270 device driver.
# 
diff -Nru a/drivers/s390/char/raw3270.c b/drivers/s390/char/raw3270.c
--- a/drivers/s390/char/raw3270.c	Tue Apr 27 04:58:57 2004
+++ b/drivers/s390/char/raw3270.c	Tue Apr 27 04:58:57 2004
@@ -381,6 +381,8 @@
 			return;	/* Sucessfully restarted. */
 		break;
 	case RAW3270_IO_STOP:
+		if (!rq)
+			break;
 		raw3270_halt_io_nolock(rp, rq);
 		rq->rc = -EIO;
 		break;
@@ -881,7 +883,7 @@
 		if (rc) {
 			/* Didn't work. Try to reactivate the old view. */
 			rp->view = oldview;
-			if (oldview->fn->activate(oldview) != 0) {
+			if (!oldview || oldview->fn->activate(oldview) != 0) {
 				/* Didn't work as well. Try any other view. */
 				list_for_each_entry(nv, &rp->view_list, list)
 					if (nv != view && nv != oldview) {
